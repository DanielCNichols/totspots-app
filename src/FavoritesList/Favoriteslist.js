import React, { Component } from 'react';
import venuesContext from '../VenuesContext';
import ApiService from '../services/api-service';
import Favorites from '../favorites/Favorites';
import './Favoriteslist.css';

export default class FavoritesList extends Component {
  static contextType = venuesContext;

  componentDidMount() {
    ApiService.getFavorites()
      .then(favorites => {
        this.context.setFavorites(favorites);
      })
      .catch(this.context.setError);
  }

  renderError() {
    let error = this.context.error
    if(this.context.error) {
      return (
        <div className="error">
          <p>Sorry something has gone wrong.{error.error}</p>
        </div>
      )
    }
  }


  prerender() {
    let { favorites, error } = this.context;
    if (favorites.length === 0) {
      return (
        <div>
          <p>You haven't saved any venues!</p>
        </div>
      );
    } 
    if (error) {
      return (
      <div role='alert'>
          {this.renderError()}
        </div>)
    }  else {
      return (
        <section>
          <header>
            <h3>Your saved venues</h3>
          </header>
          <div>
            <ul>
              {favorites.map(favorites => (
                <Favorites favorites={favorites} key={favorites.id} />
              ))}
            </ul>
          </div>
        </section>
      );
    }
  }

  render() {
    return <div>{this.prerender()}</div>;
  }
}
