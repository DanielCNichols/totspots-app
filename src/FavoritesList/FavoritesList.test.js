import React from 'react';
import ReactDOM from 'react-dom';
import FavoritesList from './FavoritesList.js';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

const favorites = [
  {
    venue_name: 'Early Bird',
    city: 'Durham',
    state: 'NC',
    address: '987 Erwin Rd.',
    zipcode: '27705',
    venue_type: 'Restaurant',
    id: 5,
    url: null,
    phone: '9848880417'
  },
  {
    venue_name: 'FullSteam',
    city: 'Durham',
    state: 'NC',
    address: '753 Parrish St. ',
    zipcode: '27705',
    venue_type: 'Bar',
    id: 33,
    url: null,
    phone: null
  }
];

it('renders FavoritesList without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FavoritesList favorites={favorites} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders FavoritesList as expected', () => {
  const tree = renderer
    .create(

        <FavoritesList name='FavoritesList' favorites={favorites}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders FavoritesList as as expected with no favorites', () => {
  const tree = renderer
    .create(

        <FavoritesList name='FavoritesList' favorites={[]}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
