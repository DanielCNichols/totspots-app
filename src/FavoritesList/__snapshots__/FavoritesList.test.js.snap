// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FavoritesList as as expected with no favorites 1`] = `
<section>
  <header>
    <h3>
      Your saved venues
    </h3>
  </header>
  <div>
    <ul />
  </div>
</section>
`;

exports[`renders FavoritesList as expected 1`] = `
<section>
  <header>
    <h3>
      Your saved venues
    </h3>
  </header>
  <div>
    <ul />
  </div>
</section>
`;
